src/components/EditCoursePage/EditCourseForm.jsx

```diff
@@ -0,0 +1,208 @@
+import React from 'react';
+import { Field, reduxForm } from 'redux-form';
+import PropTypes from 'prop-types';
+
+import RenderInputTextField from '../RenderInputTextField';
+import RenderSelectField from '../RenderSelectField';
+import ImageUpload from '../../components/ImageUpload';
+import RichEditor from '../../components/RichEditor';
+
+
+class EditCourseForm extends React.Component {
+  getOptions() {
+    if (!this.props.courseOptions) {
+      return [];
+    }
+
+    const {
+      data,
+    } = this.props.courseOptions;
+
+    if (!data || !data.actions) {
+      return [];
+    }
+
+    return data.actions.PUT;
+  }
+
+  parseOptions(inChoices) {
+    const outChoices = [];
+    for (let i = 0, len = inChoices.length; i < len; i += 1) {
+      outChoices.push({
+        label: inChoices[i].display_name,
+        value: inChoices[i].value,
+      });
+    }
+
+    return outChoices;
+  }
+
+  render() {
+    const {
+      handleSubmit,
+      number,
+      price,
+      submitting,
+    } = this.props;
+
+    const options = this.getOptions();
```

Comment:
Will this call the Get method for options on every render?

Reply:
Nope. This is just grabbing the data and assigning it to options. We fetch the options as part of the Edit Page before we do anything with the Form.